### ISSUE TITLE
POST requests failing using Amplify/api v1.0.11 

### ISSUE THREAD

**Issue by sgresh** [aws-amplify/amplify-js#1646]
**Describe the bug**
Amplify API version 1.0.11 breaks POST requests to API Gateway.
It appears v1.0.11 is adding 'content-type' twice to the signedHeaders section of the **Authorization** request header

**To Reproduce**
1. Specify "@aws-amplify/api": "1.0.11" in package.json (or use "^1.0.#")
2. Use "API.post()"
3. Request fails with a 403

**Desktop (please complete the following information):**
 - OS: Mac OSX 10.13.6
 - Browser: Chrome 68.0.3440.106

**Additional context**
The request headers coming from client, note the 'content-type' next to 'SignedHeaders', and the addition of 'charset' in the 'Content-Type' header:

Successful 200 Request headers using API v1.0.10:
```
Authorization: AWS4-HMAC-SHA256 Credential=XXXXXXXXXXXXXXX/us-west-2/execute-api/aws4_request, SignedHeaders=content-type;host;x-amz-date;x-amz-security-token, Signature=XXXXXXXXXXXXXXXXXXXX
Content-Type: application/json
```

Failing 403 Request headers when using API v1.0.11:
```
Authorization: AWS4-HMAC-SHA256 Credential=XXXXXXXXXXXXXXX/us-west-2/execute-api/aws4_request, SignedHeaders=content-type;content-type;host;x-amz-date;x-amz-security-token, Signature=XXXXXXXXXXXXXXX
Content-Type: application/json; charset=UTF-8
```

Was using "^1.0.2" in package.json:
```json
"@aws-amplify/api": "^1.0.2",
"@aws-amplify/auth": "^1.0.2",
"@aws-amplify/core": "^1.0.2",
"@aws-amplify/storage": "^1.0.2",
```

package-lock.json looked like this:
```json
  "@aws-amplify/api": {
      "version": "1.0.11",
      "resolved": "https://registry.npmjs.org/@aws-amplify/api/-/api-1.0.11.tgz",
      "integrity": "sha512-wa0v6Z/Fi3UxHDxXmBmHplRDCunglDDvpNBZn0aDUPoeW2SGKmfQLNM9RMD/xI/0V/5gnguZysHq+hMKYzJguA==",
      "requires": {
        "@aws-amplify/auth": "^1.1.2",
        "@aws-amplify/cache": "^1.0.9",
        "@aws-amplify/core": "^1.0.9",
        "@types/zen-observable": "^0.5.3",
        "axios": "^0.17.0",
        "graphql": "0.13.0",
        "zen-observable": "^0.8.6"
      },
      "dependencies": {
        "axios": {
          "version": "0.17.1",
          "resolved": "https://registry.npmjs.org/axios/-/axios-0.17.1.tgz",
          "integrity": "sha1-LY4+XQvb1zJ/kbyBT1xXZg+Bgk0=",
          "requires": {
            "follow-redirects": "^1.2.5",
            "is-buffer": "^1.1.5"
          }
        }
      }
    },
    "@aws-amplify/auth": {
      "version": "1.1.2",
      "resolved": "https://registry.npmjs.org/@aws-amplify/auth/-/auth-1.1.2.tgz",
      "integrity": "sha512-x1q/ODH9pWZuDXJ3I8GG4DzmV8Wz2jqaa5bbV3DTKkpcYzlVka6mnl0Pfwmt5qIKFDa/zGlSqMLwATICYtrsUA==",
      "requires": {
        "@aws-amplify/cache": "^1.0.9",
        "@aws-amplify/core": "^1.0.9",
        "amazon-cognito-auth-js": "^1.1.9",
        "amazon-cognito-identity-js": "^2.0.26"
      }
    },
    "@aws-amplify/cache": {
      "version": "1.0.9",
      "resolved": "https://registry.npmjs.org/@aws-amplify/cache/-/cache-1.0.9.tgz",
      "integrity": "sha512-uYRoVlriA8lMtmBfTEYyf2mTDuG8SpKhF2NiZQt0f7PtQtGwSnsIj4lagxtBSAfp7WiPzvGBH5DH0fX2L8lGFw==",
      "requires": {
        "@aws-amplify/core": "^1.0.9"
      }
    },
    "@aws-amplify/core": {
      "version": "1.0.9",
      "resolved": "https://registry.npmjs.org/@aws-amplify/core/-/core-1.0.9.tgz",
      "integrity": "sha512-tDT/t4gSmaO5dXt/BQVSbPtwRkV+AnXyJPSI3p6bfHlbNhdkVii8wqOCKfdDiLKVZdOLjd/fHNJTYYwxzvjLkA==",
      "requires": {
        "aws-sdk": "^2.312.0",
        "url": "^0.11.0"
      }
    },
    "@aws-amplify/storage": {
      "version": "1.0.9",
      "resolved": "https://registry.npmjs.org/@aws-amplify/storage/-/storage-1.0.9.tgz",
      "integrity": "sha512-mgYWdqLQL3ybZhrTnuKo9stRz4G1dRv6Vj+QyqLVomNWtJXQr5MrOxAl0S7gPwy2v0YpWKJK8fnpIkxWR+DpVg==",
      "requires": {
        "@aws-amplify/core": "^1.0.9",
        "aws-sdk": "^2.312.0"
      }
    },
```

After specifying JUST 1.0.10 for API, the POST requests worked fine:

package.json:
```json 
"@aws-amplify/api": "1.0.10",
"@aws-amplify/auth": "^1.0.2",
"@aws-amplify/core": "^1.0.2",
"@aws-amplify/storage": "^1.0.2",
```

package-lock.json after change:
```json 
  "@aws-amplify/api": {
      "version": "1.0.10",
      "resolved": "https://registry.npmjs.org/@aws-amplify/api/-/api-1.0.10.tgz",
      "integrity": "sha512-dv7x0O0WYTADPtp/onkw+WP2B61aRz3euscskJ6qnFZLh+xRAxUbNRgUxwNZYZtEQ88sh4tM9LQbO5Sr2RkQ+w==",
      "requires": {
        "@aws-amplify/auth": "^1.1.1",
        "@aws-amplify/cache": "^1.0.7",
        "@aws-amplify/core": "^1.0.7",
        "@types/zen-observable": "^0.5.3",
        "axios": "^0.17.0",
        "graphql": "0.13.0",
        "zen-observable": "^0.8.6"
      },
      "dependencies": {
        "axios": {
          "version": "0.17.1",
          "resolved": "https://registry.npmjs.org/axios/-/axios-0.17.1.tgz",
          "integrity": "sha1-LY4+XQvb1zJ/kbyBT1xXZg+Bgk0=",
          "requires": {
            "follow-redirects": "^1.2.5",
            "is-buffer": "^1.1.5"
          }
        }
      }
    },
```

My frontend code:
```javascript
let apiName = 'apiname';
let path = '/path';

let myInit = {
  headers: {
    'Content-Type': 'application/json'
  }, body: {
    object: object
  }
}
return API.post(apiName, path, myInit);
```

**Comment by elorzafe** [comment]
Thanks for creating the issue. I have reproduced the problem and I will send a PR to fix it soon.

**Comment by elorzafe** [comment]
The fix is publish on aws-amplify@unstable (1.1.4-unstable.1)

**Comment by ZebGirouard** [comment]
Thanks @elorzafe , had same issue and `1.1.5-unstable.2` upgrade worked for me!
